addons/mail/static/tests/message/message_tests.js

```diff
@@ -1148,6 +1148,98 @@ QUnit.test("allow attachment image download on message", async () => {
     await contains(".o-mail-AttachmentImage .fa-download");
 });
 
+QUnit.test("check visibility of Download All button", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "test" });
+    const [attachmentId_1, attachmentId_2, attachmentId_3] = pyEnv["ir.attachment"].create([
+        {
+            name: "test.png",
+            mimetype: "image/png",
+        },
+        {
+            name: "Blah.png",
+            mimetype: "image/png",
+        },
+        {
+            name: "shut.png",
+            mimetype: "image/png",
+        },
+    ]);
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachmentId_1, attachmentId_2],
+        body: "<p>Test</p>",
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachmentId_3],
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Message", 2);
+    await contains(".o-mail-AttachmentImage", 3);
+    const $expandButtons = $(".o-dropdown--no-caret button[title='Expand']");
+
+    // Download All button should be visible for multiple documents
+    await click($expandButtons[0]);
+    await contains(".o-dropdown--menu span[title='Download All']");
+
+    // Download All button should not be visible for a single document
+    await click($expandButtons[1]);
+    await contains(".o-dropdown--menu span[title='Download All']", 0);
+});
+
+QUnit.test("check the visibility of remove button on attachments", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "test" });
+    const [attachId_1, attachId_2] = pyEnv["ir.attachment"].create([
+        {
+            name: "attach1.txt",
+            mimetype: "text/plain",
+        },
+        {
+            name: "attach2.txt",
+            mimetype: "text/plain",
+        },
+    ]);
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachId_1],
+        body: "<p>Test</p>",
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachId_2],
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
+    pyEnv["mail.message"].create({
+        attachment_ids: [attachId_1, attachId_2],
+        model: "discuss.channel",
+        res_id: channelId,
+        message_type: "comment",
+    });
```

Comment:
Could use `pyEnv.create([])` array param to create all messages at once, so you don't have to repeat yourself with `pyEnv['mail.message'].create()`